python/ray/data/tests/test_dataset_pipeline.py

```diff
@@ -129,7 +129,11 @@ def test_split(ray_start_regular_shared):
         .map(lambda x: x + 1) \
         .repeat(10)
 
-    @ray.remote
+    # We need all the `consume` tasks to run concurrently. Since the
+    # PipelineSplitExecutorCoordinator won't move on to the next chunk until
+    # all the shards of the first split are processed. Note that the other
+    # consume tasks can't run when one of them releases its cpu.
+    @ray.remote(num_cpus=0)
```

Comment:
If we are disabling the limit for when object references are passed, can we remove the num_cpus=0 in the dataset tests?

Reply:
I don't think so. I think the issue is just that the way `dataset_pipeline.split` is implemented, all shards of the first chunk need to be read in before anyone can move on to the next chunk, so we need these tasks to run at the same time.

Reply:
Can we remove it now?